ZividOpenCV/CaptureUndistortRGB/CaptureUndistortRGB.cpp

```diff
@@ -25,32 +27,64 @@ int main()
         std::cout << "Connecting to the camera" << std::endl;
         auto camera = zivid.connectCamera();
 
-        std::cout << "Configuring the camera settings" << std::endl;
-        camera << Zivid::Settings::Iris{ 21 } << Zivid::Settings::ExposureTime{ std::chrono::microseconds{ 20000 } }
-               << Zivid::Settings::Gain{ 1 } << Zivid::Settings::Brightness{ 1.0 };
+        std::cout << "Enter <a><Return> for 2D only, and just <Return> for 3D" << std::endl;
+        const auto command = getInput();
+        bool use2D = false;
+        if(command == "a" || command == "A")
+        {
+            use2D = true;
+        }
+
+        cv::Mat rgb;
+
+        if(use2D)
+        {
+            std::cout << "2D only" << std::endl;
+
+            std::cout << "Configuring the camera settings" << std::endl;
+            auto settings = Zivid::Settings2D();
```

Comment:
This is a matter of taste, but I personally find `Zivid::Settings2D settings` easier to read. What do you think?

Reply:
Should use of auto be reserved to when the return type is somehow derived by a specific overload of a constructor? Are there any rules I should be aware of? Didn't immediately find clear guidance on when *not* to use auto.

Reply:
This is a topic of debate in the C++ community, and we don't have much about in our coding standards either. I'm fine with you using it here too, even though I wouldn't do it myself. I guess we should just agree on some guidelines.